functions/index.js

```diff
@@ -80,6 +80,42 @@ exports.getDailyMessages = functions.pubsub
             .set(parsedDailyMessages);
     });
 
+exports.getDiningInfo = functions.pubsub
+  .schedule("30 0 * * 1-5") // fetch events daily messages at 12:30am
+  .timeZone("America/New_York")
+  .onRun(async () => {
+    const diningInfo = {};
+    const promises = [];
+    const today = moment()
+      .tz("America/New_York")
+      .format("YYYY-MM-DD");
+
+    Object.keys(diningIds).forEach((id) => {
+      promises.push(getMeal(id));
+    });
+
+    await Promise.all(promises).then((values) => {
```

Comment:
you dont need this await. then will only be called when the promise.all resolves